`from` is required in the SDK but optional in the spec causing `eth_call` to often fail

## `from` is required in the SDK but optional in the spec causing `eth_call` to often fail

### Description

In the spec https://eth.wiki/json-rpc/API#eth_call and in the geth docs https://geth.ethereum.org/docs/rpc/ns-eth, the `from` parameter is marked optional, presumably for methods where the caller's data is not required (for example, retrieving a token name, retrieving a token's decimal count), but it is required here https://github.com/0xPolygon/polygon-sdk/blob/develop/jsonrpc/dispatcher.go#L405

This causes calls like this (from metamask)

```
2021-07-29T14:40:40.962-0400 [DEBUG] polygon-dev.jsonrpc: handle request: request={"id":"109b6e95-32ed-4c01-a842-3ea8ee1e5897","jsonrpc":"2.0","method":"eth_call","params":[{"to":"0xDA4e8F6A6af5909dDC6E38eA6e490099d88c0139","data":"0x95d89b41"},"0x1e"]}
```

to fail, which is actually the call to retrieve the token symbol:

```
  /**
     * @dev Returns the symbol of the token, usually a shorter version of the
     * name.
     */
    function symbol() public view returns (string memory) {
        return _symbol;
    }
```
    
SDK logs:
    
```
    2021-07-29T18:15:33.538Z [DEBUG] polygon.dispatcher: request: method=eth_call id=0b198c7b-9cf9-4373-8ae8-552805ec3d91
2021-07-29T18:15:33.538Z [ERROR] polygon.dispatcher: failed to dispatch: method=eth_call err="from is empty"
```
    


### Your environment

* OS and version Ubuntu 20 
* version of the Polygon SDK a0618c48b8896fb994f8ef42699617cb81fc4af0
* branch that causes this issue `develop`

### Steps to reproduce

* Tell us how to reproduce this issue <br />
* Where the issue is, if you know <br />
https://github.com/0xPolygon/polygon-sdk/blob/develop/jsonrpc/dispatcher.go#L405 should be marked optional. Maybe a new `decodeTxn` is needed for `eth_call` specifically? Not sure how this affects the tx execution down the line. 
* Which commands triggered the issue, if any
Going into metamask and adding a token deployed on the polygon SDK in dev mode. It'll error when trying to auto populate the token name and decimals

### Expected behaviour

Tell us what should happen

### Actual behaviour

Tell us what happens instead

### Logs

Please paste any logs here that demonstrate the issue, if they exist

### Proposed solution

If you have an idea of how to fix this issue, please write it down here, so we can begin discussing it

--- a/jsonrpc/dispatcher.go
+++ b/jsonrpc/dispatcher.go
@@ -403,7 +403,7 @@ func (d *Dispatcher) getNextNonce(address types.Address, number BlockNumber) (ui
 func (d *Dispatcher) decodeTxn(arg *txnArgs) (*types.Transaction, error) {
 	// set default values
 	if arg.From == nil {
-		return nil, fmt.Errorf("from is empty")
+		arg.From = &types.Address{}
 	}
 	if arg.Data != nil && arg.Input != nil {
 		return nil, fmt.Errorf("both input and data cannot be set")

--- a/jsonrpc/dispatcher.go
+++ b/jsonrpc/dispatcher.go
@@ -477,7 +477,7 @@ func (d *Dispatcher) getNextNonce(address types.Address, number BlockNumber) (ui
 func (d *Dispatcher) decodeTxn(arg *txnArgs) (*types.Transaction, error) {
 	// set default values
 	if arg.From == nil {
-		arg.From = &types.Address{}
+		arg.From = &types.ZeroAddress
 	}
 	if arg.Data != nil && arg.Input != nil {
 		return nil, fmt.Errorf("both input and data cannot be set")

--- a/jsonrpc/dispatcher.go
+++ b/jsonrpc/dispatcher.go
@@ -440,8 +440,7 @@ func (d *Dispatcher) decodeTxn(arg *txnArgs) (*types.Transaction, error) {
 	}
 
 	if arg.Gas == nil {
-		// TODO
-		arg.Gas = argUintPtr(1000000)
+		arg.Gas = argUintPtr(0)
 	}
 
 	txn := &types.Transaction{

--- a/jsonrpc/eth_endpoint.go
+++ b/jsonrpc/eth_endpoint.go
@@ -275,7 +275,7 @@ func (e *Eth) EstimateGas(arg *txnArgs, rawNum *BlockNumber) (interface{}, error
 	valueInt := new(big.Int).Set(transaction.Value)
 
 	// If the sender address is present, recalculate the ceiling to his balance
-	if transaction.GasPrice != nil && gasPriceInt.BitLen() != 0 {
+	if transaction.From != types.ZeroAddress && transaction.GasPrice != nil && gasPriceInt.BitLen() != 0 {
 
 		// Get the account balance
 		acc, err := e.d.store.GetAccount(header.StateRoot, transaction.From)

--- a/jsonrpc/eth_endpoint.go
+++ b/jsonrpc/eth_endpoint.go
@@ -220,6 +220,11 @@ func (e *Eth) Call(arg *txnArgs, number BlockNumber) (interface{}, error) {
 		return nil, err
 	}
 
+	// If the caller didn't supply the gas limit in the message, then we set it to maximum possible => block gas limit
+	if transaction.Gas == 0 {
+		transaction.Gas = header.GasLimit
+	}
+
 	// The return value of the execution is saved in the transition (returnValue field)
 	result, err := e.d.store.ApplyTxn(header, transaction)
 	if err != nil {

